draft-marx-qlog-event-definitions-quic-h3.md

```diff
@@ -866,41 +868,40 @@ Data:
 }
 ~~~
 
-### loss_timer_set
+### loss_timer_updated
 Importance: Extra
 
-This event is emitted when the single recovery loss timer is set.
+This event is emitted when a recovery loss timer changes state. The three main
+event types are:
+
+* set: the timer is set with a delta timeout for when it will trigger next
+* expired: when the timer effectively expires after the delta timeout
+* cancelled: when a timer is cancelled (e.g., all outstanding packets are
+  acknowledged, start idle period)
+
+Note: to indicate an active timer's timeout update, a new "set" event is used.
 
 Data:
 
 ~~~
 {
     timer_type?:"ack"|"pto", // called "mode" in draft-23 A.9.
     packet_number_space?: PacketNumberSpace,
-    timeout?:number
+
+    event_type:"set"|"expired"|"cancelled",
+
+    delta?:number // if event_type === "set": delta time in ms or us (see configuration) from this event's timestamp until when the timer will trigger
 }
 ~~~
 
 TODO: how about CC algo's that use multiple timers? How generic do these events
 need to be? Just support QUIC-style recovery from the spec or broader?
 
-Triggers:
-
-TODO
-
-### loss_timer_expired
-Importance: Extra
-
-This event is emitted when the single recovery loss timer fires.
+TODO: read up on the loss detection logic in draft-27+ and see if this suffices
 
-Data:
+Triggers:
 
-~~~
-{
-    timer_type?:"ack"|"pto", // called "mode" in draft-23 A.9.
-    packet_number_space?: PacketNumberSpace
-}
-~~~
+* no_packets_outstanding // for cancelling the timer
```

Comment:
Why do we need to export a trigger, if this is the only possible reason for canceling the timer anyway?